restore: add faq for rule not found (#16091)
pingcap/docs: faq/backup-and-restore-faq.md

- System variable tables (`mysql.tidb`, `mysql.global_variables`)
- [Other system tables](https://github.com/pingcap/tidb/blob/master/br/pkg/restore/systable_restore.go#L31)

> ### How to deal with the error of `cannot find rewrite rule` when at restoration?

Reviewer: ```suggestion
### How to deal with the error of `cannot find rewrite rule` during restoration?
```

> Please examine whether there are tables in the recovery cluster sharing the same name as other tables in the backup data but having inconsistent structures. This issue often arises due to missing indexes in the tables of the recovery cluster. A recommended approach is to delete such tables in the recovery cluster first and then proceed with the restoration.

Reviewer 2: ```suggestion
Examine whether there are tables in the restoration cluster sharing the same name as other tables in the backup data but having inconsistent structures. In most cases, this issue is caused by missing indexes in the tables of the restoration cluster. A recommended approach is to delete such tables in the restoration cluster first and then retry restoration.
```
